lib/archethic_web/controllers/api/web_hosting_controller/resources.ex

```diff
@@ -0,0 +1,180 @@
+defmodule ArchethicWeb.API.WebHostingController.Resources do
+  @moduledoc false
+
+  alias Archethic.TransactionChain.{Transaction, TransactionData}
+
+  require Logger
+
+  @spec load(tx :: Transaction.t(), url_path :: list(), cache_headers :: list()) ::
+          {:ok, file_content :: binary() | nil, encoding :: binary() | nil, mime_type :: binary(),
+           cached? :: boolean(), etag :: binary()}
+          | {:error,
+             :invalid_content
+             | :file_not_found
+             | {:is_a_directory, tx :: Transaction.t()}
+             | :invalid_encoding
+             | any()}
+  def load(
+        tx = %Transaction{
+          address: last_address,
+          data: %TransactionData{content: content}
+        },
+        url_path,
+        cache_headers
+      ) do
+    with {:ok, json_content} <- Jason.decode(content),
+         {:ok, metadata, _aeweb_version} <- get_metadata(json_content),
+         {:ok, file, mime_type, resource_path} <- get_file(metadata, url_path),
+         {cached?, etag} <- get_cache(cache_headers, last_address, url_path),
+         {:ok, file_content, encoding} <- get_file_content(file, cached?, resource_path) do
+      {:ok, file_content, encoding, mime_type, cached?, etag}
+    else
+      {:error, %Jason.DecodeError{}} ->
+        {:error, :invalid_content}
+
+      {:error, :file_not_found} ->
+        {:error, :file_not_found}
+
+      {:error, :get_metadata} ->
+        {:error, "Error: Cant access metadata and aewebversion, Reftx: #{last_address}"}
```

Comment:
Here you should use `Base.encode16` on the `last_address` otherwhise the response will be non readable